### ISSUE TITLE
Customize name of aws_iam_policy.cluster_encryption

### ISSUE THREAD

**Issue by romogo17** [terraform-aws-modules/terraform-aws-eks#1927]
Hi team, in #1915 a [new IAM Policy](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/27f99f03a7004407625f61eb030833f98326645d/main.tf#L233) was added that grants the Cluster Role permissions on the KMS used for the cluster encryption
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/27f99f03a7004407625f61eb030833f98326645d/main.tf#L230-L254

However, the name of this IAM Policy is **always based on the role name**
In my organization, there's an enforcement of different prefixes for both role and policy names. 

Would it be possible to add `cluster_encryption_policy_use_name_prefix` and `cluster_encryption_policy_name` variables (in a similar fashion to how it works for the cluster role name) for this resource as well?

```hcl
  name        = var.cluster_encryption_policy_use_name_prefix ? null : local.cluster_encryption_policy_name
  name_prefix = var.cluster_encryption_policy_use_name_prefix ? "${local.iam_role_name}-ClusterEncryption-" : null
```

**Comment by bryantbiggs** [comment]
its a role policy (i.e. - inline policy) and its pretty common/standard to name it the same as the role. However, you should be able to get around this by setting `attach_cluster_encryption_policy = false` and then attaching a policy that meets your organizations naming requirements (odd requirement but hey - to each their own)

**Comment by romogo17** [comment]
isn't the resource for inline policies [`aws_iam_role_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy) instead of [`aws_iam_policy`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_policy)?

**Comment by bryantbiggs** [comment]
ah yes, you are correct - mobile for the ~win~ failure. let me see what we can do

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.9.0 :tada:

**Comment by romogo17** [comment]
Awesome guys. Thanks!
